src/hotel-search/Merger.js

```diff
@@ -131,11 +131,13 @@ HotelSearchClient.prototype = {
               totalTaxAmountA = a.price.totalTaxAmount,
               totalTaxAmountB = b.price.totalTaxAmount;
 
-            if (totalAmountA - totalTaxAmountA === totalAmountB - totalTaxAmountB) {
-              return b.price.ecpc - a.price.ecpc;
-            } else {
-              return (totalAmountA - totalTaxAmountA) - (totalAmountB - totalTaxAmountB);
-            }
+            var basePriceA = totalAmountA - totalTaxAmountA,
+              basePriceB = totalAmountB - totalTaxAmountB;
+
+            if (basePriceA !== basePriceB) return basePriceA - basePriceB;
```

Comment:
hey @hoantony, from JIRA ticket, looks like we want to have direct partner on top disregard of the price.  Can you confirm if that is the logic? If so, comparing by provider type will be higher priority than by price.

Reply:
@viettienn the logic here is correct. We only compare the Provider_type (which DRIECT_PRIORITY has higher priority) only when the prices is in parity. So the sorting conditions will be in order: amount -> provider_type -> ecpc.